From 32e39b18fb4fd01408898a0ccba751f0033800c7 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke@hauke-m.de>
Date: Wed, 22 Jul 2015 23:05:19 +0200
Subject: [PATCH] csdk: libconnectivity_abstraction.so: add missing depending
 library

libconnectivity_abstractionso depends on libcoap.so and librt.so when
build as shared lib, which is the case for the Tizen build. This patch
adds these dependencies to the build.

Without this the dynamic loader will not automatically load these
libraries and it could result in unresolved dependencies at runtime.

Change-Id: Ib972be2b766d7fbccac9b246ae6f11e6182af94c
Signed-off-by: Hauke Mehrtens <hauke.mehrtens@lantiq.com>
Reviewed-on: https://gerrit.iotivity.org/gerrit/1825
Tested-by: jenkins-iotivity <jenkins-iotivity@opendaylight.org>
Reviewed-by: Uze Choi <uzchoi@samsung.com>
Tested-by: Uze Choi <uzchoi@samsung.com>
---
 resource/csdk/connectivity/src/SConscript |    5 +++++
 1 file changed, 5 insertions(+)

--- a/resource/csdk/connectivity/src/SConscript
+++ b/resource/csdk/connectivity/src/SConscript
@@ -108,6 +108,11 @@ if 'BLE' in ca_transport:
 print "Include path is %s" % env.get('CPPPATH')
 print "Files path is %s" % env.get('CA_SRC')
 if ca_os in ['android', 'tizen']:
+	env.AppendUnique(LIBS = ['coap'])
+	if env.get('SECURED') == '1':
+		env.AppendUnique(LIBS = ['tinydtls'])
+	if ca_os != 'android':
+		env.AppendUnique(LIBS = ['rt'])
 	calib = env.SharedLibrary('connectivity_abstraction', env.get('CA_SRC'))
 else:
 	calib = env.StaticLibrary('connectivity_abstraction', env.get('CA_SRC'))
